hamocc/mo_hamocc4bcm.F90

```diff
@@ -101,11 +105,18 @@ subroutine hamocc4bcm(kpie,kpje,kpke,kbnd,kplyear,kplmon,kplday,kldtday,pdlxp,pd
     real,    intent(out) :: pflxdms(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)      ! DMS flux [kg/m^2/s].
     real,    intent(in)  :: patmbromo(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)    ! atmospheric bromoform concentration [ppt] used in fully coupled mode.
     real,    intent(out) :: pflxbromo(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)    ! Bromoform flux [kg/m^2/s].
+    real,    intent(in)  :: patmn2o(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)      ! atmospheric nitrous oxide concentration [ppt] used in fully coupled mode.
+    real,    intent(out) :: pflxn2o(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)      ! Nitrous oxide flux [kg N2O m-2 s-1].
+    real,    intent(in)  :: patmnh3(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)      ! atmospheric ammonia concentration [ppt] used in fully coupled mode
+    real,    intent(out) :: pflxnh3(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)      ! Ammonia flux [kg NH3 m-2 s-1].
+    real,    intent(in)  :: patmnhxdep(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)   ! Atmospheric NHx deposition [kgN m-2 s-1]
+    real,    intent(in)  :: patmnoydep(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)   ! Atmospheric NOy deposition [kgN m-2 s-1]
```

Comment:
My point was just that I would suggest to avoid two (three) different variables for the same thing. Core HAMOCC does not care wether the N-deposition is read from file or coming from the atmosphere... 

I would have put the copying of atmosphere n-deposition into `ndep` into `blom2hamocc` (since it is a field that is passed through BLOM, but I'm also fine with your solution. 

Regarding whether the `read_*` routines are not entered at all or return immediately, I would vote for a consistent treatment (this is the same problem for all of these routines where an input can be switched off). I was preferring the solution with  entering and returning, because then the output field can be set to zero if needed.

Reply:
Hi @JorgSchwinger , I guess, I am a bit confused about what you mean by 2-3 different variables for the same thing... - but maybe we can quickly talk about this tmw. Wrt `blom2hamocc` - then it would make sense to do the very same for all sorts of fields coming from BLOM - e.g. also the gaseous fluxes, etc. I do see the mixture of approaches that we use, but then I would prefer to restructure all of it. Can we keep the status as is for now and merge it in (after some further testing, potentially still re-writing the reading of the N-deposition files - dependent of progress with the provision of input files)? - and we can discuss further re-organization afterward. As you might have seen, I reverted back to going in and out of the `read_ndep` (your preferred way of handling this).